roles/__bigip_selfip/tasks/main.yaml

```diff
@@ -313,6 +346,41 @@
       that:
           - not result|changed
 
+- name: Delete Self IP, specify a Route Domain
+  bigip_selfip:
+      name: "{{ selfip_name }}"
+      password: "{{ bigip_password }}"
+      server: "{{ inventory_hostname }}"
+      server_port: "{{ bigip_port }}"
+      state: "absent"
+      user: "{{ bigip_username }}"
+      validate_certs: "{{ validate_certs }}"
+      rd: "{{ selfip_rd }}"
+  register: result
+
+- name: Assert Delete Self IP
+  assert:
+      that:
+          - result|changed
+
+- name: Delete Self IP, specify a Route Domain - Idempotent check
+  bigip_selfip:
+      name: "{{ selfip_name }}"
+      password: "{{ bigip_password }}"
+      server: "{{ inventory_hostname }}"
+      server_port: "{{ bigip_port }}"
+      state: "absent"
+      user: "{{ bigip_username }}"
+      validate_certs: "{{ validate_certs }}"
+      rd: "{{ selfip_rd }}"
+  register: result
+
+- name: Assert Delete Self IP - Idempotent check
```

Comment:
The name of the assertion is just prepending the word "Assert" to the test it asserts. So this should be changed to "Assert Delete Self IP, specify a Route Domain - Idempotent check"